qlib/workflow/expm.py

```diff
@@ -267,22 +271,19 @@ def delete_exp(self, experiment_id=None, experiment_name=None):
         ), "Please input a valid experiment id or name before deleting."
         try:
             if experiment_id is not None:
-                mlflow.delete_experiment(experiment_id)
-                self.experiments = {key: val for key, val in self.experiments.items() if val.id != experiment_id}
+                self.client.delete_experiment(experiment_id)
             else:
-                experiment_id = self.experiments[experiment_name].id
-                mlflow.delete_experiment(experiment_id)
+                experiment = self.client.get_experiment_by_name(experiment_name)
+                self.client.delete_experiment(experiment.experiment_id)
         except:
             raise Exception(
                 "Something went wrong when deleting experiment. Please check if the name/id of the experiment is correct."
             )
 
     def list_experiments(self):
         # retrieve all the existing experiments
```

Comment:
We can create a constructor method to create MLflowExperiment instance from the instance created by `self.client.list_experiments`

Reply:
MLflowExperiment.create_inst_from_xxx(exp)

MLflowExperiment(exp)

Reply:
core/tools/datetimes.py: overload